Nos/Views/EditableText.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  EditableText.swift
+//  Nos
+//
+//  Created by Martin Dutra on 5/4/23.
+//
+
+import Foundation
+import SwiftUI
+import UIKit
+
+/// A ViewRepresentable that wraps a UITextView meant to be used in place of TextEditor when rich text formatting is
+/// desirable.
+struct EditableText: UIViewRepresentable {
+
+    @Binding var attributedText: AttributedString
+    @State private var selectedRange = NSRange(location: 0, length: 0)
+
+    private var font = UIFont.preferredFont(forTextStyle: .body)
+    private var insets = UIEdgeInsets(top: 16, left: 16, bottom: 16, right: 16)
+
+    init(_ attributedText: Binding<AttributedString>) {
+        _attributedText = attributedText
+    }
+
+    func makeUIView(context: Context) -> UITextView {
+        let view = UITextView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.attributedText = NSAttributedString(attributedText)
+        view.isUserInteractionEnabled = true
+        view.isEditable = true
+        view.isSelectable = true
+        view.tintColor = .accent
+        view.textColor = .secondaryTxt
+        view.font = font
+        view.backgroundColor = .clear
+        view.delegate = context.coordinator
+
+        context.coordinator.observer = NotificationCenter.default.addObserver(
```

Comment:
I think using an NSNotification for this will break if the user is using multiple windows and tabs (which we kind of accidentally support). I think we could set the `object` to `self` and then guard that `object === self` in the closure to mitigate this.

Reply:
self doesn't work because I would need to get a reference to EditableText in NewNoteView which cannot be done in swiftui. What I did was using a UUID string to mimic the same thing. I cannot test it on my mac as I cannot run nos yet, I have to update macOS

Reply:
I tried testing this on the Mac but I got a crash before I could get there :(